### PR TITLE
Move source_branch parsing into Dockerfile to allow it to work with b…

### PR DESCRIPTION
**Pull request by anuraaga**
…oth docker build and docker-compose

There isn't really a way to specify the variables available to the docker-compose tests, short of overriding the `test` hook completely. I think it's fine to move the logic for determining the version instead. This is so the `docker-*` tags can pass tests.

### CODE REVIEW

**Comment by anuraaga** [comment]
By the way, this is only if we need to push the docker images (but not zipkin artifact) again, it doesn't affect the current release of `2.18.3`

**Comment by codefromthecrypt** [comment]
ps so I know urgency, is anything broke at the moment until this? :D

File: docker/Dockerfile.release [hunk 1]
```diff
@@ -12,31 +12,35 @@
...
 
-
 # Download artifact from bintray, where the artifact should have been published by now.
 
 FROM openzipkin/zipkin-builder as built
 
-ARG version
+ARG source_branch
+
+# source_branch is something like
```
Comment (codefromthecrypt) [inline docker/Dockerfile.release:21]: good docs

**Comment by anuraaga** [comment]
The `docker-` tag, which would be used for repushing the docker image without a new java version is broken. But we don't currently need that (we'll know after checking the released docker images ;) )

File: docker/Dockerfile.release [hunk 2]
```diff
@@ -12,31 +12,35 @@
...
 # since it's where we publish to.
-
-RUN mvn org.apache.maven.plugins:maven-dependency-plugin:get \
-  -DremoteRepositories=bintray::::https://dl.bintray.com/openzipkin/maven -Dtransitive=false \
-  -Dartifact=io.zipkin:zipkin-server:${version}:jar:exec
-RUN mvn org.apache.maven.plugins:maven-dependency-plugin:get \
-  -DremoteRepositories=bintray::::https://dl.bintray.com/openzipkin/maven -Dtransitive=false \
-  -Dartifact=io.zipkin:zipkin-server:${version}:jar:slim
-
+  mvn org.apache.maven.plugins:maven-dependency-plugin:get \
```
Comment (codefromthecrypt) [inline docker/Dockerfile.release:33]: worthwhile to use the wrapper?

**Review by codefromthecrypt — APPROVED** [review]
I think I understand this. thanks!

File: docker/Dockerfile.release (hunk 2 above)
Comment (anuraaga) [reply docker/Dockerfile.release:33]: Probably is for consistency - will do it in separate PR since we should do it in the other files too.

**Pull request merged**